### PR TITLE
Prover: small-fields hotfix branch

### PR DESCRIPTION
**Pull request by gusiri**
Small-fields hot fixes after the deployment

This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] If this change is deployed to any environment (including Devnet), E2E test coverage exists or is included in this
  PR.
* [ ] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> **Medium Risk**
> Medium risk: changes prover asset serialization/loading for large limitless compiled circuits and modifies generated Solidity verifier output, both of which can break compatibility if manifests/import paths or fallback logic are incorrect. Also adjusts column naming and a few boundary-condition fixes in hashing/state-manager paths, which could affect proof generation/verification correctness.
> 
> **Overview**
> **Limitless asset I/O:** introduces a new chunked asset format (`serde.StoreChunked`/`LoadChunkedMmapBacked`) that splits large serialized objects into independently LZ4-compressed files with a `manifest`, and updates limitless asset writing/loading to prefer chunked directories for `dw-compiled-gl-*`, `dw-compiled-lpp-*`, and `dw-compiled-conglomeration` with backward-compatible `.bin` fallback.
> 
> **Solidity verifier generation:** centralizes gnark Solidity export configuration via `circuits.LineaVerifierExportOptions()` (pragma, imports, interface, chain-configuration constructor/hash/getter) and switches setup/testcase tooling to use it, simplifying `autoupdate.sh` by removing manual `sed` patching.
> 
> **Protocol & correctness fixes:** makes structured column names deterministic and caller-scoped (adds `callerName` to `NewCyclicCounter`/`CreateHeartBeat`, simplifies `RepeatedPattern`/`ManuallyShift` naming), tightens module discovery by erroring when an advice matches zero columns (plus test), normalizes/sorts trace limits in test config, and fixes several edge cases (SHA2 scan bounds/cursor advance with new tests, `codehashconsistency` loop condition for phantom ROM entries with new test, and a Keccak repeated-pattern period adjustment).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 562dc9cb2fb5ef5d04285e8456c43c223896aa2e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: prover/zkevm/limitless.go [hunk 1]
```diff
@@ -132,18 +138,21 @@ func DiscoveryAdvices(zkevm *ZkEvm) []*distributed.ModuleDiscoveryAdvice {
 		{BaseSize: 32768, Cluster: ArithOpsModuleName, Regexp: `^log[0-9]+(_u[0-9]+)?\.`},
 		{BaseSize: 32768, Cluster: ArithOpsModuleName, Regexp: `^set_byte[0-9]+\.`},
 		{BaseSize: 32768, Cluster: ArithOpsModuleName, Regexp: `^bit_xoan_u[0-9]+\.`},
-		{BaseSize: 32768, Cluster: ArithOpsModuleName, Regexp: `^mul\.`},
-		{BaseSize: 65536, Cluster: ArithOpsModuleName, Regexp: `^add\.`},
-		{BaseSize: 65536, Cluster: ArithOpsModuleName, Regexp: `^mod\.`},
+		// BaseSize increased to match limit_large to guarantee 1 segment worst case.
```
Comment (AlexandreBelling) [inline prover/zkevm/limitless.go]: Why the bumping? More than one segment worst-case is acceptable. What's not acceptable is when we have 100 segments

Comment (AlexandreBelling) [reply prover/zkevm/limitless.go]: The thing you need to understand is that when you increase a column size, this is a price you pay on all the prover requests. So just facing "1" case where this goes 4 segment on high load is not a valid reason to increase this

File: prover/protocol/dedicated/counter.go [hunk 2]
```diff
@@ -15,16 +15,18 @@ import (
...
+func NewCyclicCounter(comp *wizard.CompiledIOP, round, period int, isActiveAny any, callerName string) *CyclicCounter {
 
 	var (
 		isActive, fullyActive, size = cleanIsActive(isActiveAny)
-		numCols                     = comp.Columns.NumEntriesTotal()
-		// The appending of the round to the name is necessary to disambiguate
-		// the name of the column when serializing. This is particularly useful
-		// for self-recursion columns.
-		name = fmt.Sprintf("CYCLIC_COUNTER_%v_%v_%v", numCols, period, round)
+		// callerName disambiguates the column name independently of the number
```
Comment (AlexandreBelling) [inline prover/protocol/dedicated/counter.go]: That one might require a new setup

File: prover/protocol/dedicated/merkle/computemod.go [hunk 3]
```diff
@@ -224,8 +224,8 @@ func (cm *ComputeMod) defineNewProofAndProofEnd() {
 		panic("none of the three above cases was matched")
 	}
 
-	cm.Cols.NewProof = dedicated.CreateHeartBeat(cm.Comp, cm.Round, cm.Depth, cm.Depth-1, isActive)
-	cm.Cols.IsEndOfProof = dedicated.CreateHeartBeat(cm.Comp, cm.Round, cm.Depth, 0, isActive)
+	cm.Cols.NewProof = dedicated.CreateHeartBeat(cm.Comp, cm.Round, cm.Depth, cm.Depth-1, isActive, cm.Name+"_NEW_PROOF")
```
Comment (AlexandreBelling) [inline prover/protocol/dedicated/merkle/computemod.go:227]: Valid but requires a new setup

File: prover/config/config-sepolia-limitless.toml [hunk 4]
```diff
@@ -1,19 +1,19 @@
 environment = "sepolia"
-version = "6.2.1"                            # TODO @gbotrel hunt all version definitions.
+version = "6.2.2"                            # TODO @gbotrel hunt all version definitions.
```
Comment (AlexandreBelling) [inline prover/config/config-sepolia-limitless.toml]: Note sure why you committed that

File: prover/config/config-mainnet-limitless.toml [hunk 5]
```diff
@@ -9,13 +9,13 @@ spot_instance_reclaim_time_seconds = 120
 termination_grace_period_seconds = 1800
 
 [execution]
```
Comment (AlexandreBelling) [inline prover/config/config-mainnet-limitless.toml:11]: Not sure why you committed that

File: prover/config/config-devnet-limitless.toml [hunk 6]
```diff
@@ -11,7 +11,7 @@ prover_mode = "limitless"
 conflated_traces_dir = "./"
 requests_root_dir = "./shared/v3/prover-execution/"
 ignore_compatibility_check=false
-keep_traces_until_block = 0
+keep_traces_until_block = 9999999999
```
Comment (AlexandreBelling) [inline prover/config/config-devnet-limitless.toml]: Remove this

File: prover/zkevm/prover/statemanager/codehashconsistency/assign.go [hunk 7]
```diff
@@ -250,7 +250,21 @@ func (mod Module) Assign(run *wizard.ProverRuntime) {
 			newCRom = cRom + 1
 		}
 
-		if cSS < len(ssData) && newCRom >= len(romData) {
+		// Advance the SS cursor only when ROM was already exhausted at the
```
Comment (AlexandreBelling) [inline prover/zkevm/prover/statemanager/codehashconsistency/assign.go]: Is that just spilling a big comment?

File: prover/protocol/dedicated/counter.go [hunk 8]
```diff
@@ -24,8 +24,9 @@ func NewCyclicCounter(comp *wizard.CompiledIOP, round, period int, isActiveAny a
 		isActive, fullyActive, size = cleanIsActive(isActiveAny)
 		// callerName disambiguates the column name independently of the number
 		// of columns registered in the IOP (which varies with trace-limit
-		// scaling).
-		name = fmt.Sprintf("CYCLIC_COUNTER_%v_%v_%v", callerName, period, round)
+		// scaling). Period is intentionally excluded because it can be
+		// size-dependent (e.g. keccak COL_ROUND where period = keccakfSize).
+		name = fmt.Sprintf("CYCLIC_COUNTER_%v_%v", callerName, round)
```
Comment (AlexandreBelling) [inline prover/protocol/dedicated/counter.go:26]: Was already size independant because the period does not depend on the size

File: prover/Makefile [hunk 9]
```diff
@@ -61,7 +61,7 @@ zkevm/arithmetization/zkevm.bin: go-corset extract-constraints-git-archive
 ## Generate the setup (shared across all environments; layer2 config differentiate environments)
 ##
 setup: bin/prover
-	bin/prover setup --config ./config/config-mainnet-limitless.toml --assets-dir ./prover-assets
+	bin/prover setup --config ./config/config-sepolia-limitless.toml --circuits "execution-limitless" --assets-dir ./prover-assets
```
[the remark this answers is not in the document]
Comment (AlexandreBelling) [reply prover/Makefile]: Yes, to be reverted

File: prover/zkevm/limitless.go [hunk 10]
```diff
@@ -234,15 +244,21 @@ func DiscoveryAdvices(zkevm *ZkEvm) []*distributed.ModuleDiscoveryAdvice {
...
+		// BaseSize increased to match limit_large=2048 to guarantee 1 segment worst case.
+		{BaseSize: 2048, Cluster: TinyStuffsModuleName, Regexp: `^romlex\.`},
 		{BaseSize: 512, Cluster: TinyStuffsModuleName, Column: zkevm.StateManager.CodeHashConsistency.RomKeccak.Hi[0]},
-		{BaseSize: 2048, Cluster: TinyStuffsModuleName, Regexp: `^loginfo\.`},
-		{BaseSize: 2048, Cluster: TinyStuffsModuleName, Regexp: `^trm\.`},
+		// BaseSize set to 131072 to handle loginfo's Corset perspective expansion.
+		// loginfo columns live in hub's shared register space via XOR overlays,
+		// so the QBM column height equals the shared register size (~130K in
+		// block 29944798), NOT loginfo's own trace limit (4096/8192).
+		{BaseSize: 131072, Cluster: TinyStuffsModuleName, Regexp: `^loginfo\.`},
```
Comment (AlexandreBelling) [inline prover/zkevm/limitless.go]: tiny-stuff -> hub

File: prover/zkevm/limitless.go (hunk 1 above)
Comment (gusiri) [reply prover/zkevm/limitless.go]: reverted

File: prover/zkevm/limitless.go (hunk 10 above)
Comment (gusiri) [reply prover/zkevm/limitless.go]: moved to hub to avoid worst case 64 segments
